src/pydocstyle/checker.py

```diff
@@ -633,29 +687,84 @@ def _suspected_as_section(_line):
                                    lines[i - 1],
                                    lines[i],
                                    lines[i + 1:],
                                    i,
                                    False)
                     for i in suspected_section_indices)
 
+
         # Now that we have manageable objects - rule out false positives.
-        contexts = (c for c in contexts if self._is_a_docstring_section(c))
+        contexts = (c for c in contexts if ConventionChecker._is_docstring_section(c))
 
         # Now we shall trim the `following lines` field to only reach the
         # next section name.
         for a, b in pairwise(contexts, None):
             end = -1 if b is None else b.original_index
-            new_ctx = SectionContext(a.section_name,
+            yield SectionContext(a.section_name,
                                      a.previous_line,
                                      a.line,
                                      lines[a.original_index + 1:end],
                                      a.original_index,
                                      b is None)
-            for err in self._check_section(docstring, definition, new_ctx):
-                yield err
+
+
+    def _check_numpy_sections(self, lines, definition, docstring):
+        """D21{4,5}, D4{05,06,07,08,09,10}: Docstring sections checks.
+
+        Check the general format of a sectioned docstring:
+            '''This is my one-liner.
+
+            Short Summary
+            -------------
+            This is my summary.
+
+            Returns
+            -------
+            None.
+
+            '''
+
+        Section names appear in `NUMPY_SECTION_NAMES`.
+        """
+        for ctx in self._get_section_contexts(lines,
+                                              self.NUMPY_SECTION_NAMES):
+            yield from self._check_numpy_section(docstring, definition, ctx)
+
+    def _check_google_sections(self, lines, definition, docstring):
+        """D4{05,10,11,13,16}, D214: Section name checks.
+
+        Check the general format of a sectioned docstring:
+            '''This is my one-liner.
+
+            Note:
+                This is my summary.
+
+            Returns:
+                None.
+
+            '''
+
+        Section names appear in `GOOGLE_SECTION_NAMES`.
+        """
+        for ctx in self._get_section_contexts(lines,
+                                              self.GOOGLE_SECTION_NAMES):
+            yield from self._check_google_section(docstring, definition, ctx)
+
+    @check_for(Definition)
+    def check_docstring_sections(self, definition, docstring):
+        """Check for docstring sections."""
+        if not docstring:
+            return
+
+        lines = docstring.split("\n")
+        if len(lines) < 2:
+            return
+
+        yield from self._check_numpy_sections(lines, definition, docstring)
+        yield from self._check_google_sections(lines, definition, docstring)
```

Comment:
Using `yield from` since docs say we will drop python2.7 support from the next release.